core/creator/src/main/java/io/quarkus/creator/phase/runnerjar/RunnerJarPhase.java

```diff
@@ -309,6 +309,10 @@ public FileVisitResult visitFile(Path file, BasicFileAttributes attrs)
 
                 }
             } else {
+                // Exclude files that are not jars (typically, we can have XML files here, see https://github.com/quarkusio/quarkus/issues/2852)
+                if (!resolvedDep.getFileName().toString().endsWith(".jar")) {
```

Comment:
Shouldn't we also accept directories?

Reply:
When we build the runner jar the lib dir includes only the JARs. So, it's probably ok.

Reply:
This kind of check should move though out of the `if`. If `uberJar` is true then `resolvedDep` will be passed to `ZipUtils.newFileSystem(resolvedDep)`, which will fail if it's not an archive.

Reply:
Ah, then ball in @gsmet 's camp.

Reply:
Ah right, I had an hesitation, will fix.

Reply:
@aloubyansky done!